raft/src/test/java/org/apache/kafka/raft/MockNetworkChannel.java

```diff
@@ -25,20 +25,25 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 public class MockNetworkChannel implements NetworkChannel {
     private final AtomicInteger correlationIdCounter;
+    private final Map<Integer, InetSocketAddress> addressCache;
```

Comment:
I don't think we are using the address here. Can we use `Set<Integer>`? Potentially we could even get rid of this collection. It was more useful when the RaftClient itself was expected to discover the voter endpoints.

Reply:
On second thought, it seems worth keeping this as a set. It helps us ensure that no requests are sent to non-voters. I would just change the field name to `Set<Integer> voterIds`.

Reply:
Ack. I considered it. Figured it might be useful to have the endpoints for any future tests. Looks like the MockNetworkChannel doesn't test anything endpoint specific. Will remove